packages/govuk-frontend/src/govuk/assets/manifest.json

```
@@ -0,0 +1,19 @@
+{
+  "icons": [
```

How much do we need to match these up with **template.njk**

E.g. We've not got the 48x48 icon here (maybe that's fine)

---

I don't think they practically need to match at all. If the _only_ reason we're providing a manifest is for Android devices, we could get away with only defining the 192x192 image. 

I included the 180x180 and 512x512 as options in case a requesting service or device might find them useful, as we already have them on hand, and to provide a little bit of futureproofing in case iOS does start using the manifest (as we're seeing with macOS Sonoma), or some other usage comes along that prefers an even larger graphic to be available.

---

>…provide a little bit of futureproofing…

Yeah for this bit

i.e. For future proofing, and as a safety net, should we match **template.njk** to prepare for the day other browsers also ship support for **manifest.json**?

Making sure all the familiar sizes they support are still in there

I've double checked the [**What’s New in Safari** documentation](https://developer.apple.com/library/archive/releasenotes/General/WhatsNewInSafari/Articles/Safari_11_1.html#//apple_ref/doc/uid/TP40014305-CH14-SW6) to find when manifest support was added (Safari 11.1) but they default to `apple-touch-icon` in the HTML

Future browsers might default to `icons` in the manifest instead 🤷‍♂️

---

Sizes wise, we're all set for the [Google docs](https://web.dev/add-manifest/)

> For Chromium, you must provide at least a 192x192 pixel icon, and a 512x512 pixel icon. If only those two icon sizes are provided

Except for Android we likely want an SVG touch icons too given their examples?

>Chromium-based browsers also support SVG icons which can be scaled arbitrarily without looking pixelated and that support advanced features like [responsiveness to `prefers-color-scheme`](https://blog.tomayac.com/2021/07/21/dark-mode-web-app-manifest-app-icons/), with the caveat that the icons do not update live, but remain in the state they were in at install time.

```suggestion
  "icons": [
    {
      "src": "/images/govuk-icon.svg",
      "type": "image/svg+xml",
      "sizes": "512x512"
    },
```

---

To add, [Apple's documentation](https://developer.apple.com/design/human-interface-guidelines/app-icons) has widened since https://github.com/alphagov/govuk_template/pull/300

They suggest 1024×1024 for auto scaling, but for exact pixel sizes they list:

### iPhone
* 180×180 `@3x` — Home screen
* 120×120 `@2x` — Home screen

### iPad
* 167×167 `@1x` — Home screen (iPad Pro)
* 152×152 `@1x` — Home screen (iPad, iPad mini)

### Mac
* 512×512 `@1x` — Desktop, Dock
* 1024×1024 `@2x` — Desktop, Dock

---

We could supply all of those if we really want to, it's just not really necessary.

I've stuck with a single 180x180 that seems to be the sole size used in many modern implementations and recommended by folks dedicated enough to write about the subject.[^1][^2][^3][^4]

I'm not too concerned with re-defining every single icon from the template in the manifest either. The backwards-compatible nature of the web means that even if Safari, et al. update to _prefer_ the manifest, they're unlikely to _only_ use the manifest, and letting them choose the icon that fits best is why we specify size information in the first place.

[^1]: https://www.leereamsnyder.com/blog/favicons-in-2021
[^2]: https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs
[^3]: https://dev.to/masakudamatsu/favicon-nightmare-how-to-maintain-sanity-3al7
[^4]: realfavicongenerator.net changed to only supply a single 180x180 icon way back in 2016 — https://github.com/RealFaviconGenerator/realfavicongenerator/issues/211

---

There's only one size missing if we say the desktop sizes are out of scope?

* 120×120 `@2x` — Home screen

Don't have to add the "desktop" sizes, maybe the SVG though? https://github.com/alphagov/govuk-frontend/pull/4185#discussion_r1324567044

---

I've opted to relent and add all of the icons to the manifest. Seemingly this isn't unexpected (MDN has examples that include even the .ico being defined within the manifest), but I'll be surprised if at least some of these aren't superfluous, being unused by any current browser/device.

I'm not sure whether the current SVG supplied is really suitable for the intended use cases either. It's not the same design as the PNGs offered and doesn't come with any included margins, so if it were to be selected for an app icon on a device it may end up looking a bit weird. Something to test.